piperider_cli/initializer.py

```diff
@@ -58,6 +58,7 @@ def _ask_user_input_datasource(config: Configuration = None):
             console.print(
                 f'[[bold yellow]Warning[/bold yellow]] No credential found for \'{ds.type_name}\' datasource \'{ds.name}\'')
             if _ask_user_update_credentials(ds):
+                config.dump(PIPERIDER_CONFIG_PATH)
```

Comment:
Under this condition, the `config.yml` file should already exist. We only ask users to fill in credentials data. If we dump the config file again, it could override the comments added by users in the `config.yml` file.

Reply:
We revert the behavior back to only dumping `credential.yml`.
Because the `config.yml` is source controlled and should not be overwritten.